Source/Core/Scheduler/L4StateMachine/StateMachine.cpp

```diff
@@ -106,6 +106,8 @@ bool StateMachine::ExportData(StructuredDataI & data) {
 bool StateMachine::Initialise(StructuredDataI &data) {
     ErrorManagement::ErrorType err;
     err.parametersError = !ReferenceContainer::Initialise(data);
+    StreamString firstState;
+    data.Read("FirstState", firstState);
```

Comment:
Return value is not being checked.
What happens in the FirstState is not specified (and it should be an optional parameter)? I suspect that currentState would not be initialised. 

A possible solution is to always initialise `currentState = Get(0)` and then loop and update as you propose below.

Please also add the relevant unit tests that demonstrate the behaviour and update the StateMachine.h with the new parameter.